crates/dns-vcr/src/lib.rs

```diff
@@ -0,0 +1,452 @@
+//! Test utility to record and replay DNS transactions with internet-based servers.
+
+use std::{
+    env::{self, VarError},
+    fs::{File, OpenOptions},
+    io::{self, ErrorKind, Write},
+    net::{Ipv4Addr, SocketAddr, UdpSocket},
+    path::PathBuf,
+    string::FromUtf8Error,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    thread::{self, JoinHandle},
+    time::{Duration, Instant},
+};
+
+use serde::{Deserialize, Serialize};
+use tracing::{debug, error};
+
+mod wireshark;
+
+/// Receive timeout used when listening on the mock server.
```

Comment:
Nit: prefer a top-down topological ordering, high-level public API first, then have these `const`s at the bottom (before `mod tests`). `Error` should thus also be closer to the bottom.